### PR TITLE
WT-10591 Automate LazyFS testing using Evergreen

### PR DESCRIPTION
**Pull request by pmacko86**
This PR adds LazyFS testing to Evergreen, including some code changes to make it work.

The biggest one there was moving the FIFO file out to the `/tmp` directory (which I explained in one of the comments), plus a few changes to help ensure that that file gets cleaned up in most circumstances. I'm not quite sure how to make it 100% foolproof, so if you have any ideas, please don't hesitate to share :)

The second question is the frequency of these tests, I'm starting them off with once per day, but we can easily change it...

### CODE REVIEW

File: test/evergreen_develop.yml [hunk 1]
```diff
@@ -133,3 +133,20 @@ buildvariants:
       batchtime: 40320 # 28 days
     - name: s-outdated-fixmes
       batchtime: 10080 # 7 days
+
+- name: infrequent-tests-lazyfs
+  display_name: "~ Infrequent tests with LazyFS"
```
Comment (lukech) [inline test/evergreen_develop.yml:138]: Just hope to point out that we don't have to create a separate build variant these days to achieve a different running frequency for given tasks. Evergreen now supports the task-level setting of test frequency (`batchtime` as used below) to override the build-variant-level setting. We could consider adding these 3 new tasks to the existing `ubuntu2004` builder. If the aim is to make LazyFS have its own builder for ease of identification please feel free to ignore my suggestion.

File: test/evergreen_develop.yml [hunk 2]
```diff
@@ -133,3 +133,20 @@ buildvariants:
...
+    cmake_generator: Ninja
+    num_jobs: $(echo "`grep -c ^processor /proc/cpuinfo` * 2" | bc)
+  stepback: false
+  tasks:
+    - name: csuite-random-abort-lazyfs
+      batchtime: 1440 # 1 day
+    - name: csuite-schema-abort-lazyfs
+      batchtime: 1440 # 1 day
+    - name: csuite-timestamp-abort-lazyfs
+      batchtime: 1440 # 1 day
```
Comment (lukech) [inline test/evergreen_develop.yml:152]: If all tasks share the same `batchtime` value, this setting can be put at the builder-level (and set once).

File: test/evergreen_develop.yml (hunk 1 above)
Comment (lukech) [reply test/evergreen_develop.yml:138]: Now I see there's a special cmake setting below (on line 142) to enable LazyFS, it makes sense to have a separate build variant (builder).

**Review by lukech — APPROVED** [review]
LGTM % nit

**Review by keitharnoldsmith — APPROVED** [review]
LGTM.

A minor point you might want to clean up is that the calls to memset the WT_LAZY_FS structure to zero aren't needed. It looks like these programs all declare it as a global variable, and C guarantees/requires that global variables are automatically initialized to zero.

**Comment by pmacko86** [comment]
Thank you both for the approvals and the feedback!

In response to your comments, I did:
* Moved `batchtime` to builder level.
* Removed the unnecessary `memset`'s.
* Renamed "Infrequent tests with LazyFS" to "LazyFS tests" (this is in response to a Slack exchange with Keith).

**Comment by pmacko86** [comment]
evergreen merge

**Pull request merged**